commands/azureCommands/delete-registry.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import { SubscriptionModels } from "azure-arm-resource";
+import * as vscode from "vscode";
+import { dockerExplorerProvider } from '../../dockerExtension';
+import { AzureRegistryNode } from '../../explorer/models/azureRegistryNodes';
+import * as acrTools from '../../utils/Azure/acrTools';
+import { AzureUtilityManager } from '../../utils/azureUtilityManager';
+import { confirmUserIntent, quickPickACRRegistry } from '../utils/quick-pick-azure';
+
+/** Delete a registry and all it's associated nested items
+ * @param context : the AzureRegistryNode the user right clicked on to delete
+ */
+export async function deleteAzureRegistry(context?: AzureRegistryNode): Promise<void> {
+    let registry: Registry;
+    if (context) {
+        registry = context.registry;
+    } else {
+        registry = await quickPickACRRegistry(false, 'Select the registry you want to delete');
+    }
+    const shouldDelete = await confirmUserIntent(`Are you sure you want to delete ${registry.name} and its associated images? Enter yes to continue: `);
```

Comment:
nit: You should let confirmUserIntent() append the 'Enter Yes to continue' portion to the prompt, since it's the one that's expecting that answer from the user anyway.